pkgs/by-name/si/simplemoji/package.nix

```diff
@@ -0,0 +1,64 @@
+{
+  autoPatchelfHook,
+  darwin,
+  fetchFromGitHub,
+  lib,
+  libxkbcommon,
+  fontconfig,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  wayland,
+  xorg,
+  noto-fonts-color-emoji,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "simplemoji";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "SergioRibera";
+    repo = "simplemoji";
+    rev = "v${version}";
+    hash = "sha256-MU31gRikMqXelWjwjMPZtfbDpybYlPTFr1u9w2843Kc=";
+  };
+
+  cargoHash = "sha256-NvNNc+V7wloPxM532uNrXoou4vnKLhFK29L2/17eB98=";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    pkg-config
+  ];
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.AppKit
+      darwin.apple_sdk.frameworks.CoreFoundation
+      darwin.apple_sdk.frameworks.CoreGraphics
+      darwin.apple_sdk.frameworks.Foundation
+      darwin.apple_sdk.frameworks.OpenGL
+    ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    stdenv.cc.cc.libgcc
+    fontconfig.dev
+    libxkbcommon.dev
+    wayland
+    xorg.libxcb
+    xorg.libX11
+  ];
+
+  runtimeDependencies = [
+    noto-fonts-color-emoji
```

Comment:
We are `patchelf` it into the package?? If you need it, you probably want to put it somewhere in the package, or offer it by environment variables set in a wrapper.

Reply:
I mean, the fonts package is just fonts, it's not a library. It makes no sense to put it in there. The `autoPatchelfHook` needs to be put in `nativeBuildInputs` of course

Reply:
@Aleksanaa I'm moved `autoPatchelfHook` as runtime dependency, thanks for the suggestion

Reply:
> I mean, the fonts package is just fonts, it's not a library. It makes no sense to put it in there.

I'm afraid not, the application doesn't embed the source in the final binary, it loads it at runtime so the sources are as a runtime dependency, maybe a good idea is to add a configuration parameter so they can change the source, but I don't know how to do that.

> The `autoPatchelfHook` needs to be put in `nativeBuildInputs` of course

My mistake, I'll fix it in a couple of hours please

Reply:
Some things you can't judge by name. The role of runtimeDependencies here is consumed by autoPatchelfHook.

Reply:
There's no "runtime dependencies" like traditional package managers. If you want your result to depend on another output path, you have to associate it with that output path (Which usually means you need to include it explicitly).

Reply:
@Aleksanaa 
> Some things you can't judge by name. The role of runtimeDependencies here is consumed by autoPatchelfHook.

Okay, I got it all wrong.

> There's no "runtime dependencies" like traditional package managers.

So, it seems to me that in a new version I will have to give support to some parameter to assign custom fonts, for the moment how can I do so that whenever you use the `simplemoji` package also install the `noto-fonts-color-emoji`?

Reply:
If you are not playing tricks on searching fonts (well if you definitely want to bind the font, you should), it uses fontconfig. There should be some fontconfig environment variables, and you can wrap your binary with it. But it seems that other emoji pickers don't specially handle this, so they actually don't care about whether they can be displayed or not.

Or maybe you need to use a set of SVG to display emojis instead of directly displaying fonts.

Reply:
@Aleksanaa That would require a change in the way the application works, I am willing to do it, but I would like to know if there is any option I can do so that this PR can be merged and in a next version make the change.

Would removing the `noto-fonts-color-emoji` dependency be enough?